### DISCUSSION TITLE
Retrieve checked value in edit form

### DISCUSSION THREAD

**Discussion by AlexWS1048575** [adonisjs/core#4267] [Help]
I would like to retrieve checked value in AdonisJS, but I don't know how to retrieve checked value when I click edit
My table
`users`
```
id | username
1 | 987654
6 | 123456
```
`permissions`
```
id | name
1 | View Permissions
2 | Edit Permissions
3 | Add Permissions
4 | Delete Permissions
```

`users_permissions`
```
id | user_id | permission_id
1 | 6 | 1
2 | 6 | 3
```

My `edit.edge` file
```
@each (groupedPermissions  in permgroup)
  <div>
  &nbsp; <span class="text-primary"><input type="checkbox" id="selectGroupPermissions" ><strong> Group: {{ groupedPermissions.group }}</strong></span>
    @each (permission in permissions.filter(p => p.group === groupedPermissions.group))
      <div>
        <label>
          &nbsp; &nbsp; <input type="checkbox" id="permissions[]" name="permissions[]" value="{{ permission.id }}" {{ userPermissions.permission_id ? 'checked' : '' }}>
          {{ permission.name }} - {{ permission.description }}
        </label>
      </div>
    @endeach
  </div>
@endeach
```

My `UsersController.ts` file
```
public async edit({ view, params }: HttpContextContract) {
    const users    = await User.findOrFail(params.id);
    const permissions = await `Database.from('permissions').select('*')`
    const userPermissions = await users.related('permissions').query()
    const permgroup = await Permission.query().distinct('group')

    return view.render('users/edit', { users: users, permissions: permissions, userPermissions: userPermissions, permgroup: permgroup })
  }
```

I've trying to using pluck but it return an error `"Property 'pluck' does not exist on type 'DatabaseQueryBuilderContract<any>'.` In older version, it works. How to retrieve a checked value like this image everytime I click edit users? I try this code in top it won't work.

![laravel pluck](https://github.com/adonisjs/core/assets/101855082/e76e0b73-28a0-4468-96f0-c187f443222e)

**Comment by tomgobich** [comment]
First, the error is correct, AdonisJS 5 doesn't have a `pluck` method, though it does still have `select` which will limit the actual query to only querying the selected column(s), on models though, to match the expected TypeScript return type, it'll still be sent back as a model instance.

Second, if I'm understanding correctly, you're just trying to get `userPermissions` to be an array of id's the user has permissions for, correct? 

If so, you have a couple of options.

```ts
// loop over the returned items to grab just the selected column
const userPermissionIds = (await user.related('permissions').query().select('id')).map(user => user.id)

// query your pivot table directly
const userPermissionIds = await Database
  .from('users_permissions')
  .where('user_id', user.id)
  .select('permission_id')
```

Or, you could even add in your own macro to the query builder that would allow you to do something like:
```ts
const userPermissionIds = await user.related('permissions').query().selectIds()
```
You can find more info on that approach here if you'd like: https://adocasts.com/lessons/my-favorite-adonisjs-model-query-builder-macros
